File: solution/122BestTimeToBuyAndSellStockII/122BestTimeToBuyAndSellStockII.go
package main

func maxProfit(prices []int) int {
	n := len(prices)
	profit := 0
	for i := 1; i < n; i++ {
		diff := prices[i] - prices[i-1]
		if diff > 0 {
			profit += diff
		}
	}
	return profit
}

func main() {

}

File: src/122BestTimeToBuyAndSellStockII/122BestTimeToBuyAndSellStockII.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut profit = 0;
        for i in 0..prices.len() as i32 - 1 {
            profit += std::cmp::max(0, prices[i as usize + 1] - prices[i as usize]);
        }
        profit
    }
}
